packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -1164,6 +1166,8 @@ export class ProjectLifecycleManager {
   }
 
   setConfigFilePath (fileName: string) {
+    this._configFile = fileName
```

Comment:
This feels like duplication. Do we really need another source of truth when we already have `_configFilePath`? Should we just derive this?

```ts
get configFile () {
  return this.ctx.modeOptions.configFile ?? path.basename(this.configFilePath) ?? 'cypress.config.js'
}